### PR TITLE
Cache empty arrays

### PR DESCRIPTION
**Pull request by khellang**
Quick structural search & replace with R# :smile:

### CODE REVIEW

**Comment by jchannon** [comment]
What does this solve?

**Comment by khellang** [comment]
> What does this solve?

It "solves" a bunch of allocations :smile:

File: src/Nancy/ArrayCache.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿namespace Nancy
+{
+    public class ArrayCache
```
Comment (thecodejunkie) [inline src/Nancy/ArrayCache.cs]: You know better than to try and sneak undocumented classes below my radar :stuck_out_tongue_closed_eyes:

File: src/Nancy/ArrayCache.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+{
+    public class ArrayCache
+    {
+        public static T[] Empty<T>()
+        {
+            return EmptyArray<T>.Value;
+        }
+    }
+
+    internal static class EmptyArray<T>
```
Comment (thecodejunkie) [inline src/Nancy/ArrayCache.cs]: Maybe stick this as a nested class of `ArrayCache` ?

**Comment by thecodejunkie** [comment]
Neat little trick :+1:

**Pull request merged**